### PR TITLE
Separate AdaptiveApplication2 part from feature/machineStateManagerIm…

### PR DESCRIPTION
**Pull request by rBekesh**
…prov.

Code of AdaptiveApplication1 was reused in proc2 and proc3 of AdaptiveApplication2 instead of creating separate similar class.

Removed SimpleApplication

### CODE REVIEW

File: applications/AdaptiveApplication1/src/aa_main.cpp [hunk 1]
```diff
@@ -27,6 +29,8 @@ int main()
 
 static void signalHandler(int signo)
 {
-    std::cout << "received signal:" << sys_siglist[signo] << "\n";
+    std::cout << "[ " << appName << " ]:\tReceived signal: "
```
Comment (adklimkiew) [inline applications/AdaptiveApplication1/src/aa_main.cpp]: appName prefix should be somehow added automatically

File: applications/AdaptiveApplication1/include/adaptive_app.hpp [hunk 2]
```diff
@@ -12,7 +12,7 @@ class State;
 class AdaptiveApp
 {
 public:
-    AdaptiveApp(std::atomic<bool>& terminate);
+    AdaptiveApp(std::atomic<bool>& terminate, const std::string appName);
```
Comment (adklimkiew) [inline applications/AdaptiveApplication1/include/adaptive_app.hpp]: const string&

File: applications/AdaptiveApplication2/src/proc2.cpp [hunk 3]
```diff
@@ -1,23 +1,36 @@
-#include <iostream>
-#include <thread>
+#include <adaptive_app.hpp>
+#include <state.hpp>
```
Comment (adklimkiew) [inline applications/AdaptiveApplication2/src/proc2.cpp]: State header needed?

File: applications/AdaptiveApplication2/src/proc3.cpp [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
+        std::cout << "[ " << appName << " ]:\tError while registering signal." << std::endl;
+    }
+
+    AdaptiveApp app(isTerminating, appName);
+
+    while (true)
     {
-        std::cout << "proc3" <<std::endl;
-        std::this_thread::sleep_for(std::chrono::seconds(1));
+        app.transitToNextState();
```
Comment (adklimkiew) [inline applications/AdaptiveApplication2/src/proc3.cpp]: I thought this function is private / protected...

File: applications/AdaptiveApplication1/include/adaptive_app.hpp [hunk 5]
```diff
@@ -6,7 +6,10 @@
 #include <i_application_state_client_wrapper.h>
 
 #include <vector>
+#include <memory>
+#include <atomic>
 
+#include <application_state_client.h>
```
Comment (ulianaStashkiv) [inline applications/AdaptiveApplication1/include/adaptive_app.hpp]: move it after #include <i_application_state_client_wrapper.h>

File: applications/AdaptiveApplication2/src/proc2.cpp [hunk 6]
```diff
@@ -1,23 +1,37 @@
...
+
+    AdaptiveApp app2(std::make_unique<StateFactory>(),
+                     std::make_unique<api::ApplicationStateClientWrapper>());
+
+    app2.init();
+
+    while (!isTerminating)
+    {
+        app2.run();
+        std::this_thread::sleep_for(std::chrono::seconds(5));
```
Comment (ulianaStashkiv) [inline applications/AdaptiveApplication2/src/proc2.cpp]: magic number

File: applications/AdaptiveApplication2/src/proc3.cpp [hunk 7]
```diff
@@ -1,23 +1,37 @@
...
-	return 0;
+    AdaptiveApp app3(std::make_unique<StateFactory>(),
+                     std::make_unique<api::ApplicationStateClientWrapper>());
+
+    app3.init();
+
+    while (!isTerminating)
+    {
+        app3.run();
+        std::this_thread::sleep_for(std::chrono::seconds(5));
```
Comment (ulianaStashkiv) [inline applications/AdaptiveApplication2/src/proc3.cpp]: magic number

**Pull request merged**